app/Http/Controllers/Api/ProjectController.php

```diff
@@ -44,88 +59,152 @@ public function add_project(Request $request){
-            $project->project_address_id = $request->project_address_id;
+            if (!$project) {
+                return response()->json(['message' => 'Project not found'], 404);
+            }
             $project->forename = $request->forename;
             $project->surname = $request->surname;
             $project->project_name = $request->project_name;
             $project->description = $request->description;
             $project->contact_mobile_no = $request->contact_mobile_no;
+            $project->contact_home_phone = $request->contact_home_phone;
             $project->contact_email = $request->contact_email;
+            $project->postcode = $request->postcode;
+            $project->county = $request->county;
+            $project->town = $request->town;
+            $project->categories = $request->categories;
+            $project->subcategories = $request->subcategories;
+            $project->notes = $request->notes;
             if(!$project->save()){
                 return response()->json(['message'=>'Unexpected error, please try after sometimes'],400);
             }
+            //delete images
+            $previousFiles = Projectfile::where('project_id', $project->id)->get();
+
+            foreach ($previousFiles as $previousFile) {
+                $s3Path = 'Testfolder/' . $previousFile->filename;
+                Storage::disk('s3')->delete($s3Path);
+                $previousFile->delete();
+            }
+            //update images
+            foreach ($request->images as $file) {
+                $fileName = $file->getClientOriginalName();
+                $extension = $file->getClientOriginalExtension();
+                $s3FileName = Str::uuid().'.'.$extension;
+                $file_type = explode('/', mime_content_type($file->getRealPath()))[0];
+                Storage::disk('s3')->put('Testfolder/'.$s3FileName, file_get_contents($file->getRealPath()));
+                $path = Storage::disk('s3')->url('Testfolder/'.$s3FileName);
+
+                $projectfile_entry = Projectfile::updateOrCreate(
+                    ['project_id' => $project->id],
+                    [
+                        'file_type' => $file_type,
+                        'filename' => $s3FileName,
+                        'file_original_name' => $fileName,
+                        'file_extension' => $extension,
+                        'url' => $path
+                    ]
+                );
+            }
             return response()->json(['message'=>"Project updated successfully"], 200);
         } catch(Exception $e){
+            DB::rollback();
             return response()->json($e->getMessage(), 500);
         }
     }
 
-    public function get_file(Request $request){
+    public function cancel_project(Request $request){
         $validator = Validator::make($request->all(), [
-            'project_id' => 'required|Project',
+            'project_id' => 'required',
+            'status' => 'required'
```

Comment:
Status won't be send from the app if you are creating two APIs for cancelling and pausing project

Reply:
resolved